src/ensembl/src/content/app/browser/browserState.ts

```diff
@@ -18,10 +18,16 @@ import browserStorageService from './browser-storage-service';
 
 import { BrowserTrackStates } from './track-panel/trackPanelConfig';
 
-const activeGenomeId = browserStorageService.getActiveGenomeId();
-const activeEnsObjectIds = browserStorageService.getActiveEnsObjectIds();
-const trackStates = browserStorageService.getTrackStates();
-const chrLocations = browserStorageService.getChrLocation();
+const isServer = typeof window === 'undefined';
+
+const activeGenomeId = !isServer
+  ? browserStorageService.getActiveGenomeId()
+  : null;
```

Comment:
Here and below, wouldn't it be better to move the `isServer` logic inside the getters itself?

Reply:
If you mean, move the` isServer` inside the storage services, then we will lose the type predictability of the getters — effectively, all their return values will become potentially `undefined`. However, storage services shouldn't even be run on the server — it doesn't make sense, at least as long as we keep all our data client-side, to do so. As far as a storage service is concerned, it is always run in the browser. So I am reluctant to move the server/client check into the storage services.